Mathlib/GroupTheory/GroupAction/Group.lean

```diff
@@ -178,6 +180,26 @@ theorem smul_eq_iff_eq_inv_smul (g : α) {x y : β} : g • x = y ↔ x = g⁻¹
 
 end Group
 
+section Monoid
+
+variable [Monoid α] [MulAction α β]
+
+variable {c : α} {x y : β} [Invertible c]
+
+@[simp]
+theorem invOf_smul_smul : ⅟c • c • x = x := inv_smul_smul (unitOfInvertible c) _
+
+@[simp]
+theorem smul_invOf_smul : c • (⅟ c • x) = x := smul_inv_smul (unitOfInvertible c) _
```

Comment:
`c` and `x` should be explicit for these lemmas.

Reply:
Okay, done. Please can you explain why though so I know for next time.

Reply:
The simple answer here is that this lemma is almost identical to `smul_inv_smul`, and so you should try to match the argument explicitness to.

The slightly more general answer is that lemmas with `=` in their goal should ensure that every variable in their goal is captured in an an explicit variable (either as `(x : X)` or `{x : X} (hx : P x)`).